### PR TITLE
UCT/SELF: Fix MD config table

### PR DESCRIPTION
**Pull request by dmitrygladkov**
## What

Fix MD config table set for SELF

## Why ?

Wrong config table set

## How ?

Replace IFACE config table set for MD component by empty MD config table

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10223/ for details (Mellanox internal link).

**Comment by dmitrygladkov** [comment]
@yosefe this fixes MD config used by me in #3568. could you review pls?

File: src/uct/sm/self/self.c [hunk 1]
```diff
@@ -398,4 +398,4 @@ static UCT_MD_COMPONENT_DEFINE(uct_self_md, UCT_SELF_NAME,
                                uct_self_query_md_resources, uct_self_md_open, NULL,
                                uct_self_md_rkey_unpack,
                                ucs_empty_function_return_success, "SELF_",
-                               uct_self_iface_config_table, uct_self_iface_config_t);
+                               uct_md_config_table, uct_self_iface_config_t);
```
Comment (brminich) [inline src/uct/sm/self/self.c]: looks like `uct_self_iface_config_t` -> `uct_md_config_t` needed as well

Comment (dmitrygladkov) [reply src/uct/sm/self/self.c]: good catch, I didn't notice this. fixed

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10246/ for details (Mellanox internal link).

**Pull request merged**